### PR TITLE
#929 Add LeetCode icon

### PR DESCRIPTION
**Pull request by TheKodingKlingon**
Adding the icon for #929

### CODE REVIEW

**Comment by ericcornelissen** [comment]
Is this an official SVG, or did you create it based on the PNG on the page you linked @TheKodingKlingon? Also, can you tell me where you got the color from? Other than that the PR looks great

**Comment by TheKodingKlingon** [comment]
Thanks @ericcornelissen, the icon comes specifically from [here](https://leetcode.com/subscribe/) and is indeed a converted png. I've dug through the site looking for svg although none are present. The colour is a primary colour present throughout the site and is the only non-black/white colour present in all iconography.

**Comment by ericcornelissen** [comment]
I think the icon looks good @TheKodingKlingon, but I'll wait for another maintainer to have a look as well before merging this.

Regarding the color, I'm finding `#FEA116` and `color: orange` (which for me translates to `#FFA500`, but this differs may differ from browser to browser) rather then `#FFA300`

**Comment by birjolaxew** [comment]
Some minor conversion artifacts; I believe the line ends are supposed to be nice clean halfcircles, which they are not in the SVG:

<p align="center"> <code>...</code> </p>

As for the colors, I'm getting:

- `#F89F1B` from [their primary logo PNG](https://leetcode.com/static/images/LeetCode_logo_rvs.png)
- `#D8890C` from [their small logo PNG](https://leetcode.com/static/webpack_bundles/images/LeetCode_nav.4d940ca7.png)
- `#FEA116` from their CSS (e.g. `.navitem-highlight` or `.price-panel .header`) 
- [`#FFA500`](https://www.w3.org/TR/css-color-4/#named-colors) from their CSS (e.g. `.new`)

In other words they aren't remotely close to being consistent themselves. I'd suggest using `#F89F1b` simply because it is what they use for their primary logo.

**Comment by birjolaxew** [comment]
Here's a cleaned up version of the SVG - if you could add this (or a similar clean SVG) and update the color, the PR is good to go :)

<p align="center"></p>

```svg
<svg aria-labelledby="simpleicons-leetcode-icon" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title id="simpleicons-leetcode-icon">LeetCode icon</title><path d="M16.102 17.93l-2.697 2.607c-.466.467-1.111.662-1.823.662s-1.357-.195-1.824-.662l-4.332-4.363c-.467-.467-.702-1.15-.702-1.863s.235-1.357.702-1.824l4.319-4.38c.467-.467 1.125-.645 1.837-.645s1.357.195 1.823.662l2.697 2.606c.514.515 1.365.497 1.9-.038.535-.536.553-1.387.039-1.901l-2.609-2.636a5.055 5.055 0 0 0-2.445-1.337l2.467-2.503c.516-.514.498-1.366-.037-1.901-.535-.535-1.387-.552-1.902-.038l-10.1 10.101c-.981.982-1.494 2.337-1.494 3.835 0 1.498.513 2.895 1.494 3.875l4.347 4.361c.981.979 2.337 1.452 3.834 1.452s2.853-.512 3.835-1.494l2.609-2.637c.514-.514.496-1.365-.039-1.9s-1.386-.553-1.899-.039z"/><path d="M20.811 13.01H10.666c-.702 0-1.27.604-1.27 1.346s.568 1.346 1.27 1.346h10.145c.701 0 1.27-.604 1.27-1.346s-.569-1.346-1.27-1.346z"/></svg>
```

**Comment by birjolaxew** [comment]
:+1: Thanks for the contribution! It is much appreciated.
